flocker/ca/test/test_ca.py

```diff
@@ -21,6 +23,297 @@
+        ``NodeCredential.from_path`` cannot be opened for reading.
+        """
+        path = FilePath(self.mktemp())
+        path.makedirs()
+        # random UUID that doesn't exist in path
+        uuid = self.uuid
+        cert_file = b"{uuid}.crt".format(uuid=uuid)
+        key_file = b"{uuid}.key".format(uuid=uuid)
+        crt_path = path.child(cert_file)
+        crt_file = crt_path.open(b'w')
+        crt_file.write(b"dummy")
+        crt_file.close()
+        key_path = path.child(key_file)
+        key_file = key_path.open(b'w')
+        key_file.write(b"dummy")
+        key_file.close()
+        # make file unreadable
+        key_path.chmod(64)
+        e = self.assertRaises(
+            PathError, NodeCredential.from_path, path, uuid
+        )
+        expected = (
+            b"Private key file could not be opened. "
+            b"Permission denied {path}"
+        ).format(path=key_path.path)
+        self.assertEqual(str(e), expected)
+
+    def test_certificate_subject_node_uuid(self):
+        """
+        A cert written by ``NodeCredential.initialize`` has the
+        subject common name "node-{uuid}" where {uuid} is the UUID
+        generated during the certificate's creation.
+        """
+        nc = NodeCredential.initialize(self.path, self.ca)
+        cert = nc.certificate.original
+        subject = cert.get_subject()
+        self.assertEqual(subject.CN, b"node-{uuid}".format(uuid=nc.uuid))
+
+    def test_certificate_ou_matches_ca(self):
+        """
+        A cert written by ``NodeCredential.initialize`` has the issuing
+        authority's common name as its organizational unit name.
+        """
+        nc = NodeCredential.initialize(self.path, self.ca)
+        cert = nc.certificate.original
+        issuer = cert.get_issuer()
+        subject = cert.get_subject()
+        self.assertEqual(
+            issuer.CN,
+            subject.OU
+        )
+
+    def test_certificate_is_signed_by_ca(self):
+        """
+        A cert written by ``NodeCredential.initialize`` is validated
+        as being signed by the certificate authority.
+        """
+        nc = NodeCredential.initialize(self.path, self.ca)
+        cert = nc.certificate.original
+        issuer = cert.get_issuer()
+        self.assertEqual(
+            issuer.CN,
+            self.ca.certificate.getSubject().CN
+        )
+
+    def test_certificate_expiration(self):
+        """
+        A cert written by ``NodeCredential.initialize`` has an expiry
+        date 20 years from the date of signing.
+        """
+        today = datetime.datetime.now()
+        expected_expiry = today + datetime.timedelta(seconds=EXPIRY_20_YEARS)
+        nc = NodeCredential.initialize(self.path, self.ca)
+        cert = nc.certificate.original
+        asn1 = cert.get_notAfter()
+        expiry_date = datetime.datetime.strptime(asn1, "%Y%m%d%H%M%SZ")
+        self.assertEqual(expiry_date.date(), expected_expiry.date())
```

Comment:
Another branch fixes this expiry date issue across all the ``flocker.ca` tests. See #1339